libraries/AP_MotionController/AP_MotionController.cpp

```diff
@@ -45,13 +45,36 @@ AP_MotionController::~AP_MotionController()
 void AP_MotionController::init(void)
 {
     AP_SerialManager &serial_manager = AP::serialmanager();
-    port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Volz, 0);
+    port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_ESCTelemetry, 0);
+    roboclaw.init(port, 10000);
     //hal.console->printf("AP_MotionController initialized.\n");
     //gcs().send_text(MAV_SEVERITY_INFO, "AP_MotionController Initialized.");
 }
 
-void AP_MotionController::update(void)
+double AP_MotionController::constrain_map(double x, double in_min, double in_max, double out_min, double out_max)
 {
-    //hal.console->printf("AP_MotionController updated.\n");
-    //gcs().send_text(MAV_SEVERITY_INFO, "AP_MotionController Updated.");
+    //constrain the input value between in_min and in_max
+    if (x < in_min)
+        x = in_min;
+    else if (x > in_max)
+        x = in_max;
+    //map the value from the scale (in_min->in_max) into scale (out_min->out_max)
+    return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+}
+
+void AP_MotionController::update(uint16_t steering, uint16_t throttle)
+{
+    static int counter = 0;
+    uint32_t m1 = AP_MotionController::constrain_map(steering, SRV_Channels::srv_channel(1)->get_output_min(), SRV_Channels::srv_channel(1)->get_output_max(), 0, 127);
+    uint32_t m2 = AP_MotionController::constrain_map(steering, SRV_Channels::srv_channel(3)->get_output_min(), SRV_Channels::srv_channel(3)->get_output_max(), 0, 127);
+    roboclaw.ForwardBackwardM1(AP_RoboClaw::ROBOCLAW_ADDR, m1);
+    roboclaw.ForwardBackwardM2(AP_RoboClaw::ROBOCLAW_ADDR, m2);
+    //roboclaw.SpeedM1M2(AP_RoboClaw::ROBOCLAW_ADDR, steering, throttle);
+    if (++counter >= 10) // Call every fifth of a second (50Hz calling rate)
+    {
+        hal.console->printf("Time:%05.3f Steering:%d Throttle:%d\n", AP_HAL::millis64() / 1000.0, m1, m2);
+        //port->printf("Time:%05.3f Steering:%d Throttle:%d\n", AP_HAL::millis64()/1000.0, steering, throttle);
```

Comment:
It's best to remove commented out code.  It just becomes messy if we leave this in.  For debug, you'll find examples where other libraries have a DEBUG_OUTPUT (or something like that) if you really want to leave the debug in there for future development.  Personally I'd say just remove it though.